### PR TITLE
Add dense option to `categorical_column_with_vocabulary_list` Feautre Column

### PR DESCRIPTION
**Pull request by yupbank**
Right now categorical_column_with_vocabulary_list only provides SparseTensor. It's hard for people to use it with tfrecords and Canned estimator. labels have to be Tensor and no one is converting the sparse string tensor into string tensor anywhere

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @rohan100jain: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by hgadig** [comment]
@rohan100jain  Could you PTAL and approve.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @rohan100jain: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @rohan100jain: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 29 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @rohan100jain: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 44 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @rohan100jain: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 59 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/python/feature_column/feature_column.py [hunk 1]
```diff
@@ -1332,6 +1333,8 @@ def _categorical_column_with_vocabulary_list(key,
       Must be castable to `dtype`.
     dtype: The type of features. Only string and integer types are supported.
       If `None`, it will be inferred from `vocabulary_list`.
+    shape: The shape of the featutres. If `None`, the result will be
```
Comment (lamberta) [inline tensorflow/python/feature_column/feature_column.py]:
```suggestion
    shape: The shape of the features. If `None`, the result will be
```

File: tensorflow/python/feature_column/feature_column_v2.py [hunk 2]
```diff
@@ -1711,6 +1712,8 @@ def categorical_column_with_vocabulary_list(key,
       Must be castable to `dtype`.
     dtype: The type of features. Only string and integer types are supported. If
       `None`, it will be inferred from `vocabulary_list`.
+    shape: The shape of the featutres. If `None`, the result will be
```
Comment (lamberta) [inline tensorflow/python/feature_column/feature_column_v2.py]:
```suggestion
    shape: The shape of the features. If `None`, the result will be
```

**Comment by gbaned** [comment]
@yupbank - Hi, could you please look into these build failures? Thanks!

**Comment by gbaned** [comment]
@yupbank Did you get a chance to look on build failures? Please let us know on the update. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**